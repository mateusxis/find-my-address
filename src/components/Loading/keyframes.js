import { keyframes } from 'styled-components';

export const rotate = keyframes`
  from {
    transform: rotate(0deg);
  } to {
    transform: rotate(360deg);
  }
`;

export const appear = keyframes`
  from {
    opacity: 0;
  } to {
    opacity: 1;
  }
`;
