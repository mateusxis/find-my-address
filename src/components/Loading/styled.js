import styled from 'styled-components';
import { appear, rotate } from './keyframes';

export const SLoading = styled.div`
  animation: ${appear} 500ms backwards;
  position: relative;

  &:before,
  &:after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: transparent;
    border: 0.3em solid #42424220;
    border-radius: 100%;
  }

  &:after {
    border-radius: 100%;
    border: 0.3em solid #424242;
    border-left-color: transparent;
    border-top-color: transparent;
    border-bottom-color: transparent;
    animation: ${rotate} 800ms linear infinite;
  }

  width: 2rem;
  height: 2rem;

  @media (min-width: 601px) {
    height: 2.4rem;
    width: 2.4rem;
  }

  @media (min-width: 961px) {
    height: 2.6rem;
    width: 2.6rem;
  }

  @media (min-width: 1281px) {
    width: 2.8rem;
    height: 2.8rem;
  }
`;
