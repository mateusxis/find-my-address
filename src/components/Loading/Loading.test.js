import renderer from 'react-test-renderer';

import Loading from './Loading';

describe('Loading', () => {
  it('should render Loading', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
