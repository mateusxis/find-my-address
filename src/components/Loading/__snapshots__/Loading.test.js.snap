// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render Loading 1`] = `
<div
  className="sc-bdvvtL iiXgSl"
/>
`;
